bin/geninfo

```diff
@@ -2296,15 +2337,18 @@ sub intermediate_json_to_info($$$)
 
 			$branch_num = 0;
 			# Branch data
-			if ($br_coverage && !$brexcl->{$line}) {
+			if ($br_coverage && (!defined($brexcl->{$line}) || ($brexcl->{$line} != 1))) {
 				for my $b (@$branches) {
 					my $brcount = $b->{"count"};
+					my $is_exception = $b->{"throw"};
 
-					if (!defined($brcount) || $unexec) {
-						$brcount = "-";
+					if (!$is_exception || ((!defined($brexcl->{$line}) || ($brexcl->{$line} != 2)) && ($br_coverage != 2))) {
+						if (!defined($brcount) || $unexec) {
```

Comment:
Please try to limit the line length - strict 80 characters is not necessary, but don't go that far beyond that limit.

Reply:
Except for a couple of long regex'es which are difficult to split over several lines I think I've reduced line size properly.